cpp/include/cudf/utilities/traits.hpp

```diff
@@ -474,7 +475,7 @@ template <typename T>
 constexpr bool is_rep_layout_compatible()
 {
   return cudf::is_numeric<T>() or cudf::is_chrono<T>() or cudf::is_boolean<T>() or
-         cudf::is_byte<T>();
+         cudf::is_byte<T>() or std::is_same_v<T, char>;
```

Comment:
This seems a little hacky. I think I'd prefer casting `head()` rather than allowing someone calling `data<char>()`.

Reply:
Right. This is a hack.
will replace `data<char>()` with static cast `head()` everywhere.

Reply:
This hack is removed.